lib/mocha.js

```diff
@@ -866,3 +872,46 @@ Mocha.prototype.run = function(fn) {
 
   return runner.run(done);
 };
+
+/**
+ * Assigns hooks to the root suite
+ * @param {MochaRootHookObject} [hooks] - Hooks to assign to root suite
+ * @chainable
+ */
+Mocha.prototype.rootHooks = function rootHooks(hooks) {
+  if (utils.type(hooks) === 'object') {
+    var beforeAll = [].concat(hooks.beforeAll || []);
+    var beforeEach = [].concat(hooks.beforeEach || []);
+    var afterAll = [].concat(hooks.afterAll || []);
+    var afterEach = [].concat(hooks.afterEach || []);
```

Comment:
I haven't understood this part. Why you pass an object, not a function.
`Suite.prototype.beforeAll = function(title, fn)` expects a function, but here it gets an object.
If you run those hooks with `Runnable#run`, how can this work? The hook has already been executed and contains the return value, not the function anymore?

Reply:
`hooks` is a collection of all four types of hook, and has properties `beforeAll`, `beforeEach`, `afterAll`, `afterEach`.  At this point in the code, if any of these property is truthy, they are expected to be arrays of functions.  I think I could be more specific about the shape of the object; before we reach this point, `loadHooks()` (in `run-helpers.js`, iirc) does some pre-processing.

Below, we loop thru the properties (again, which are arrays of functions) and assign them to the root suite via the appropriate method in `Suite`.